### PR TITLE
is_docker_running raises instead of returning False 

### PR DESCRIPTION
**Pull request by i-a-andrianov**
when no client given and no Docker available

### CODE REVIEW

File: src/ebonite/build/docker.py [hunk 1]
```diff
@@ -151,7 +149,7 @@ def is_docker_running(client=None) -> bool:
         else:
             client.images.list()
         return True
-    except (ImportError, requests.exceptions.ConnectionError, errors.DockerException):
+    except:  # noqa
```
Comment (mike0sv) [inline src/ebonite/build/docker.py:152]: why not just add that exception? imho except without specifying exceptions is acceptable only of you reraise exception after

Comment (i-a-andrianov) [reply src/ebonite/build/docker.py]: As for your question how could we be sure that our list is full, especially taking into account that `docker` library could raise anything (nothing is documented and could change during updates)? We just create unnecessary coupling to libraries like `requests` here and make our code fragile.

NB: I changed this line to `except Exception` not to catch things like `SystemExit`, etc. This doesn't make sense for sure.

Comment (mike0sv) [reply src/ebonite/build/docker.py]: https://realpython.com/the-most-diabolical-python-antipattern/
docker stuff should only raise `DockerException` as it says in code
`ConnectionError` is derived from IOError - we can use it to remove `requests` import
but to finally decide I want to know which exception was thrown in the first place?

Comment (i-a-andrianov) [reply src/ebonite/build/docker.py]: Thanks for awesome clickbait title. You won: I hardcoded list of exceptions back (replaced `requests` with `IOError`).

The problem was caused by `RuntimeError` in `create_docker_client` when calling `is_docker_running()`. For some reasons methods `is_docker_running` and `create_docker_client` recursively call each other.

Comment (mike0sv) [reply src/ebonite/build/docker.py]: this conversation was not in vein - this is actually a bug which I added in previous PR :(. I will fix it myself in this PR, if you dont mind

**Pull request merged**